libmscore/scale.h

```diff
@@ -0,0 +1,50 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2002-2012 Werner Schweer
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#ifndef __SCALE_H__
+#define __SCALE_H__
+
+/**
+ \file
+ Definition of Scale class.
+ */
+
+#include "note.h"
+
+namespace Ms {
+
+class Scale {
+public:
+      bool loadScale(const QString&);
+      double getTuning(const Note* note) const;
+
+private:
+      QString getNextLine(QTextStream& in);
+
+      std::vector<double> originalNotes;
+      std::map<Tpc, double> computedNotes;
+
```

Comment:
"_the difference is so small that I don't mind storing double._"

The difference is small memory-wise; less small CPU-wise... Not to mention the 'illusory precision' effect: it is far from certain that the final audible result would be precise even to the cent (with the usual sampling rates around 44Kz, an A at 880Hz only has 50 samples to work with, assuming the synthesizer does not add its own curses), to make worth storing, computing, displaying _n_ decimals.

The original plug-in for ver. 1.x can be found at https://musescore.org/en/project/scales

Please note, though, that I now have better ideas about the features the editor should have to be really useful.